### ISSUE TITLE
Testing of seekrets rulesets using BATS

### ISSUE THREAD

**Issue by LinuxBozo** [18F/laptop#74]
In order to ensure that our regex rules are correct, they should be tested, using [BATS](https://github.com/sstephenson/bats) and run automatically with Travis.

Rulesets for [`git-seekrets`](https://github.com/apuigsech/git-seekret) are defined as regex inside a yaml file. These include rule name, the regex for the `match`, and any acceptable false positives as `unmatch`.  Existing rulesets that need tests defined are here:
https://github.com/18F/laptop/tree/seekret/seekret-rules

Depends on #75 for easier implementation. 

If fixing, please use `seekret` branch so it's merged into PR #69 

More about the rules YAML: https://github.com/apuigsech/seekret/blob/75ab481819b4c1e1794497defea9a9332943295b/README.rst#rules

#### Acceptance Criteria

- [x] Install `git seekret` in a repeatable way ( should be covered in another story )
- [x] Install `BATS` in a repeatable way, [readme](https://github.com/sstephenson/bats)
- [x] Write `BATS` tests for each of these:
    - [x] Check for `git seekret` installation.
    - [x] Check for `rulesets` enabled.
    - [x] Check that `git seekret` is called on a test repository and secrets are detected. (positives)
    - [x] Check that `git seekret` is called on a test repository and non-secrets are not detected. (acceptable false positives)

**Comment by pburkholder** [comment]
Could you elaborate on what sort of testing you'd like to see here? That executing a set of git commands results in an error?

I might be able to help.

P

**Comment by LinuxBozo** [comment]
@pburkholder yeah, I was capturing these quickly as I was going through things, and planned on actually providing the meat soonish..

**Comment by alain-hoang** [comment]
Working to resolve dependency on issue #75

**Comment by alain-hoang** [comment]
This still requires missing dependency on #75 before it can pass tests.  I've added pushed a branch with [here](https://github.com/18F/laptop/tree/seekrets-bats) with current WIP for tests.

**Comment by alain-hoang** [comment]
Missing dependency on #75 has been updated however it appears that git-seekret is not properly applying rulesets on staged files or commits.  Created [#22](https://github.com/apuigsech/git-seekret/issues/22) to address this

**Comment by rogeruiz** [comment]
The test repo can be created manually every time.

```
git init
touch README.md
echo "SHHHH... Secrets in this file" >> README.md
echo "fb^!uQ2oYEVPqfwSA=VP=6CXY9O8yYeIMcX8FcidX@CMP6O%Jv*9*MI+D+rZj!7a" >> README.md
echo "192.168.11.0" >> README.md
git commit # this should trigger the secrets and this is what we want to test in `BATS`.
```

**Comment by alain-hoang** [comment]
Instructions for how to run bats tests in `seekrets.bats` test file

**Comment by ertzeid** [comment]
It looks like all of the items are checked. Is this done?

**Comment by LinuxBozo** [comment]
In discussing with @alain-hoang this is still waiting on https://github.com/apuigsech/git-seekret/pull/26

**Comment by LinuxBozo** [comment]
Closed by #87 and #94
